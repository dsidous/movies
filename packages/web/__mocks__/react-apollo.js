const apollo = require.requireActual('react-apollo');
const result = jest.fn(Component => Component);

apollo.graphql = jest.fn().mockReturnValue(result);

module.exports = apollo;
