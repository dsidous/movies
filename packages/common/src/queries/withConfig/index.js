import { graphql } from 'react-apollo';
import { query } from './query';

const withConfig = () =>
  graphql(query, {
    options: () => ({
      fetchPolicy: 'cache-first',
    }),
    props: ({ data: { config = {}, loading } }) => ({
      configLoading: loading,
      config,
    }),
  });

export default withConfig;
