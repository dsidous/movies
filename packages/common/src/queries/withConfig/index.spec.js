import { graphql } from 'react-apollo';
import React from 'react';
import { shallow } from 'enzyme';
import withConfig from '.';

jest.mock('react-apollo', () => ({
  ...jest.requireActual('react-apollo'),
  graphql: jest.fn(),
}));

describe('Queries/withConfig', () => {
  graphql.mockImplementation((_, config) => {
    const ownProps = {};
    const props = config.props({
      data: {
        config: {
          images: {
            base_url: 'path/to/image',
            secure_base_url: 'path/to/image',
            backdrop_sizes: ['1', '2', '3', '4'],
            poster_sizes: ['1', '2', '3', '4'],
            profile_sizes: ['1', '2', '3', '4'],
            still_sizes: ['1', '2', '3', '4'],
          },
        },
        loading: true,
      },
      ownProps,
    });
    const options = config.options(ownProps);
    return jest.fn(Mock => <Mock options={options} {...props} />);
  });

  const MockComponent = props => ({ ...props });
  const Component = withConfig()(MockComponent);

  it('should render as expected', () => {
    const render = shallow(Component);
    expect(render).toMatchSnapshot();
  });
});
