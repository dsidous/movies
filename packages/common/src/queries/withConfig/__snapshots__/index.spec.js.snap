// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries/withConfig should render as expected 1`] = `ShallowWrapper {}`;
